generic-server/src/plugin/container.rs

```diff
@@ -89,6 +88,35 @@ impl<'a> PluginContainer<'a> {
         }
     }
 
+    pub fn unload(&mut self) -> Result<(), PluginContainerError> {
+        // Unmount plugins and clearing the vector of routes
+        let errors: HashMap<_, _> = self
+            .plugins
+            .iter()
+            .filter_map(|plugins| match plugins.unmount() {
+                Ok(_) => {
+                    tracing::info!("Unmounted plugins {}", plugins.name());
+                    self.collected_routes.clear();
+                    None
```

Comment:
Like I said above, the lambda function iterates through the plugins one by one. It is therefore not logical to clear ALL the routes on the first plugin of the stream. Move that operation below, maybe right before flagging the container as unloaded.

Reply:
This actually makes a lot of sense